lxd/api_1.0.go

```diff
@@ -163,12 +179,64 @@ func api10Get(d *Daemon, r *http.Request) Response {
 		Kernel:             kernel,
 		KernelArchitecture: kernelArchitecture,
 		KernelVersion:      kernelVersion,
-		Storage:            d.Storage.GetStorageTypeName(),
-		StorageVersion:     d.Storage.GetStorageTypeVersion(),
 		Server:             "lxd",
 		ServerPid:          os.Getpid(),
 		ServerVersion:      version.Version}
 
+	// COMMENT(brauner): Get a list of all storage drivers currently in use
+	// on this LXD instance. Only do this when we do not already have done
+	// this once to avoid unnecessarily querying the db. All subsequent
+	// updates of the cache will be done when we create or delete storage
+	// pools in the db. Since this is a rare event, this cache
+	// implementation is a classic frequent-read, rare-update case so
+	// copy-on-write semantics without locking in the read case seems
+	// appropriate. (Should be cheaper then querying the db all the time,
+	// especially if we keep adding more storage drivers.)
+	if !storagePoolDriversCacheInitialized {
+		tmp, err := dbStoragePoolsGetDrivers(d.db)
+		if err != nil && err != NoSuchObjectError {
+			return InternalError(err)
+		}
+
+		storagePoolDriversCacheMu.Lock()
+		storagePoolDriversCacheVal.Store(tmp)
+		storagePoolDriversCacheMu.Unlock()
+
+		storagePoolDriversCacheInitialized = true
+	}
+
+	drivers := readStoragePoolDriversCache()
+	// COMMENT(brauner): Currently no storage pools are configured/no
+	// drivers are in use.
+	nDrivers := len(drivers)
+	if nDrivers == 0 {
+		env.Storage = ""
+		env.StorageVersion = ""
+	}
+
+	for i := 0; i < nDrivers; i++ {
```

Comment:
With the above removed, change that to the much simpler:

    for _, driver := range drivers {